music-player/src/components/SongsList.tsx

```diff
@@ -0,0 +1,72 @@
+import { List, Button } from 'antd';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+
+import { songsNamesList } from '../data';
+import { State } from '../types';
+import { AddSongToPlaylist } from './AddSongToPlaylist';
+import { showDeleteConfirm } from './RemoveSongFromPlaylist';
+import { getPlaylists, getCurrentPlaylistName, getCurrentSongIndex } from '../reducers';
+
+const SongsList = ({
+  songsToDisplay,
+  currentSongIndex,
+  dispatch,
+  songsIndices,
+  currentPlaylistName,
+}: {
+  songsToDisplay: string[];
+  currentSongIndex: number;
+  dispatch: Dispatch;
+  songsIndices: number[];
+  currentPlaylistName: string;
+}) => {
+  const songToComponent = (song: string, indexInPlaylist: number) => {
```

Comment:
You should pass to this component the "real" index since it's completely unaware of the playlist.